### PR TITLE
Feature/gui output console

### PR DESCRIPTION
**Pull request by riedersa**
#122

### CODE REVIEW

File: source/ui/Components/Output/Console.qml [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+                text = "";
+            }
+
+        }
+
+        Connections {
+            target: outputComponent
+            // Send when the memory changes (at any address).
+            onMemoryChanged: {
+                //console.log("onMemoryChanged");
```
Comment (goldsborough) [inline source/ui/Components/Output/Console.qml]: Debug output

File: source/ui/Components/Output/Console.qml [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+                var _baseAddress = outputComponent.getOutputItem(outputItemIndex)["baseAddress"];
+                var _mode = outputComponent.getOutputItem(outputItemIndex)["textMode"];
+                // Check if the memory address that was changed (at least partly) belongs to
+                // the output item's source space.
+                if ((address+length) >= _baseAddress && _mode == 0) {
+                    it.updateContent(_baseAddress);
+                }
+                else if(_mode != 0 && (address == _baseAddress|| (address<_baseAddress && address+length >= _baseAddress)) ){
```
Comment (goldsborough) [inline source/ui/Components/Output/Console.qml]: put the `else if` on the same line as the closing brace. Also, this condition is too long. Use a boolean variable or check it inside if clause (i.e. only do `if _mode != 0 {` and inside there the second check)

File: source/ui/Components/Output/Console.qml [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+                    it.updateContent(_baseAddress);
+                }
+            }
+            // Send when any item's settings where updated.
+            onOutputItemSettingsChanged: {
+                //console.log("onOutputItemSettingsChanged");
+                var mode = outputComponent.getOutputItem(outputItemIndex)["textMode"];
+                var baseAddress = outputComponent.getOutputItem(outputItemIndex)["baseAddress"];
+                if(mode == oldMode && baseAddress == oldAddress){
```
Comment (goldsborough) [inline source/ui/Components/Output/Console.qml]: use `===`

File: source/ui/Components/Output/ConsoleSettingsWindow.qml [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+    height: 200
+    property int outputItemIndex: 2
+
+    title: "Console Settings"
+
+    // Refreshes the window's control contentItem.
+    function updateSettings() {
+        baseAddressTextField.text = outputComponent.getOutputItem(outputItemIndex)["baseAddress"];
+        var t = outputComponent.getOutputItem(outputItemIndex)["textMode"];
+        if(t==0){
```
Comment (goldsborough) [inline source/ui/Components/Output/ConsoleSettingsWindow.qml]: `===`

File: source/ui/Components/Output/ConsoleSettingsWindow.qml [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+            ab.checked = true;
+        }
+        else{
```
Comment (goldsborough) [inline source/ui/Components/Output/ConsoleSettingsWindow.qml]: ` } else {`

File: source/ui/output-component.cpp [hunk 6]
```diff
@@ -92,3 +93,33 @@ QList<bool> OutputComponent::getMemoryContent(int address, int length) const {
   }
   return contentList;
 }
+
+QString OutputComponent::getTextFromMemory(int start, QString currentText, int mode){
+    std::string text ="";
+    if(mode ==0/*ArrayBased*/){
+        for(int i=0; (start+i)<_memoryAccess.getMemorySize().get() ; i++){
```
Comment (goldsborough) [inline source/ui/output-component.cpp]: Please format this...

File: source/ui/output-component.cpp [hunk 7]
```diff
@@ -92,3 +93,33 @@ QList<bool> OutputComponent::getMemoryContent(int address, int length) const {
...
+            MemoryValue m = _memoryAccess.getMemoryValueAt(start+i).get();
```
Comment (goldsborough) [inline source/ui/output-component.cpp]: no single letter variable names

File: source/ui/Components/Output/ConsoleSettingsWindow.qml [hunk 8]
```diff
@@ -0,0 +1,165 @@
...
+                }
+
+
+                Component.onCompleted: {
+                    settingsWindowC.updateSettings();
+                }
+
+                // Reads the current input and passes the new value to the model.
+                function processInput() {
+                    var inputValue;
```
Comment (goldsborough) [inline source/ui/Components/Output/ConsoleSettingsWindow.qml]: Could maybe use the ternary operator here: `var inputValue = ab.checked ? 0 : 1;` or even `!ab.checked`

File: source/ui/Components/Output/ConsoleSettingsWindow.qml [hunk 9]
```diff
@@ -0,0 +1,165 @@
...
+                }
+            }
+
+            // Text field for settings the number of light-strips.
+            RowLayout {
+                id: mode
+                ExclusiveGroup { id: modeGroup }
+
+                RadioButton{
+                    id:ab
```
Comment (goldsborough) [inline source/ui/Components/Output/ConsoleSettingsWindow.qml]: space

**Review by goldsborough — APPROVED** [review]

**Review by FelixOpolka — APPROVED** [review]
I have reviewed your code and everything looks good!

**Review by goldsborough — APPROVED** [review]

**Pull request merged**